ImputationPipeline/Imputation.wdl

```diff
@@ -0,0 +1,475 @@
+      bcftools index -f valid_variants.bcf 
+    echo false > valid_file.txt
+    fi
+  >>>
+  output {
+    File valid_chunk_bcf ="valid_variants.bcf"
+    File valid_chunk_bcf_index = "valid_variants.bcf.csi"
+    Boolean valid = read_boolean("valid_file.txt")
+  }
+  runtime {
+    docker: "farjoun/impute:0.0.4-1506086533"
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "4 GB"
+  }
+}
+
+task PrePhaseVariantsEagle {
+  input {
+    File dataset_bcf
+    File dataset_bcf_index
+    File reference_panel_bcf
+    File reference_panel_bcf_index
+    String chrom
+    Boolean valid_chunk
+    String genetic_map_file
+    Int start
+    Int end
+  }  
+  Int disk_size = 400
+  command <<<
+    if [ ~{valid_chunk} = "true" ]; then
+      /eagle  \
+             --vcfTarget ~{dataset_bcf}  \
+             --vcfRef ~{reference_panel_bcf} \
+             --geneticMapFile ~{genetic_map_file} \
+             --outPrefix pre_phased_~{chrom} \
+             --vcfOutFormat z \
+             --bpStart ~{start} --bpEnd ~{end} --allowRefAltSwap 
+    else 
+        bcftools view ~{dataset_bcf} -Oz -o pre_phased_~{chrom}.vcf.gz
+    fi
+  >>>
+  output {
+    File dataset_prephased_vcf="pre_phased_~{chrom}.vcf.gz"
+  }
+  runtime {
+    docker: "skwalker/imputation:test"
+      memory: "32 GB"
+      cpu: "8"
+      disks: "local-disk " + disk_size + " HDD"
+  }
+}
+
+task minimac4 {
+  input {
+    File ref_panel
+    File phased_vcf
+    String prefix
+    String chrom
+    Int start
+    Int end
+    Boolean valid_phased_vcf
+  }
+  command <<<
+    if [ ~{valid_phased_vcf} = "true" ]; then
+    /Minimac4 --refHaps ~{ref_panel} --haps ~{phased_vcf} --start ~{start} --end ~{end} --window 500000 \
+      --chr ~{chrom} --noPhoneHome --format GT,DS,GP --allTypedSites --meta --prefix ~{prefix} --minRatio 0.00001 
+    else 
+      cat ~{phased_vcf} > ~{prefix}.dose.vcf.gz
+      echo "\n" > ~{prefix}.info
+    fi
+  >>>
+  output {
+    File vcf = "~{prefix}.dose.vcf.gz"
+    File info = "~{prefix}.info"
+  }
+  runtime {
+    docker: "skwalker/imputation:test"
+    memory: "4 GB"
+      cpu: "1"
+        disks: "local-disk 100 HDD"
+  }
+}
+
+task MergeVCFs {
+  input {
+    Array[File] input_vcfs
+    String output_vcf_basename
+    Float shard_size
+  }
+  Int disk_size = ceil(((shard_size * 22) * 2) + 20)
+  command <<<
+    valid_vcfs=$(for file in ~{sep=' ' input_vcfs}; do rows=$(zgrep -v "#" $file | wc -l); if [ $rows -gt 0 ]; then echo $file; fi; done)
+    bcftools concat $valid_vcfs | bgzip -c > ~{output_vcf_basename}.vcf.gz
+  >>>
+  runtime {
+    docker: "farjoun/impute:0.0.3-1504715575"
+    memory: "3 GiB"
+    disks: "local-disk " + disk_size + " HDD"
+  }
+  output {
+    File output_vcf = "~{output_vcf_basename}.vcf.gz"
+  }
+}
+
+task UpdateHeader {
+  input {
+    File vcf
+    File ref_dict
+    Int disk_size = 400
+  }
+  command <<<
+  zgrep -v "##contig=<ID=" ~{vcf} | bgzip -c > no_contigs.vcf.gz 
```

Comment:
why is this needed?  Isn't the `UpdateVCFSequenceDictionary` line just going to remove and replace all these anyway?